### ISSUE TITLE
SS4 compatability

### ISSUE THREAD

**Issue by sparkcom** [sheadawson/silverstripe-zenvalidator#64]
many thanks for your work.

any plan to upgrade it to SS4 ready?

**Comment by sheadawson** [comment]
I don't have time for it at the moment but happy to merge any PRs or give full access to anyone who wants to do the upgrade. Thanks!

**Comment by lekoala** [comment]
I've just pushed to master preliminary support for ss4. Basic usage is covered.

I have other changes in mind as well, to be more in line with modern standards. This will break a few things.

Also, it's worth mentioning this module that uses ParsleyJS and supports ss4 as well
https://github.com/praxisnetau/silverware-validator

**Comment by sheadawson** [comment]
Thanks @lekoala. Silverware validator looks pretty solid and probably a good replacement for this module seeing as I'm not really maintaining it anymore. I might put a note about that in the readme, just redirect new users to silverware validator. What do you think? Do you see yourself wanting to continue using/working on this module or switch to the Silverware one?

**Comment by lekoala** [comment]
Yes, Silverware validator seems to work great and I've already added a link in the readme. I've just updated this module to ease migration to ss4 for existing projects, so that everything remains compatible.
It would make sense to focus on improving only one module, and Silverware seems to be a better place to start for ss4. I'll keep making a few tweaks for ss4 so that this module works fine, but I think we should make it clear that no big changes are coming and that people should look for alternatives if they can.
